### PR TITLE
feat(resp): add option to disable XML serialization of errors

### LINKED ISSUE #2355: Add flag to disable XML serialization; deprecate `HTTPError.to_xml()`
This serialization is kinda it's own thing that's different from everything else falcon behaves wrt media serialization.

Also in 2024 it's unlikely that people writing python application will ever touch xml

### PR DESCRIPTION
**Pull request by CaselIT**
Fixes #2355

### CODE REVIEW

**Comment by vytas7** [comment]
Not directly related to this PR, but it seems we also need to update the wording of docs for [`App.set_error_serializer()`](https://falcon.readthedocs.io/en/latest/api/app.html#falcon.App.set_error_serializer) regarding media handlers.

File: falcon/response.py [hunk 1]
```diff
@@ -1387,26 +1387,43 @@ class ResponseOptions:
...
+    Default handlers are provided for the ``application/json``,
     ``application/x-www-form-urlencoded`` and ``multipart/form-data`` media types.
     """
     static_media_types: Dict[str, str]
     """A mapping of dot-prefixed file extensions to Internet media types (RFC 2046).
 
     Defaults to ``mimetypes.types_map`` after calling ``mimetypes.init()``.
     """
+    enable_xml_error_serialization: bool
+    """When ``False`` disables support for serializing http errors using XML when using
```
Comment (vytas7) [inline falcon/response.py:1399]: Suggested rewording
```rst
"""Set to ``False`` to disable automatic inclusion of the XML handler in the default error serializer (:arrow_left: link to its docs)
(default ``True``).

Enabling this option does not automatically render all error response in XML,
but only the client prefers (via the ``Accept`` request header) XML to JSON and other configured media handlers (link to docs?).
The browser, however, uses to prefer XML; so use this option exclude it from the supported media types.

Note:
    This option will be removed in Falcon 5.0 (with XML error serialization disabled by default).

Note:
    This option has no effect when a custom error serializer (link to docs?) is in use.
```

**Review by vytas7 — CHANGES_REQUESTED** [review]
Looks good, just the docs need a bit polish, and I would prefer another name for the option itself (but I don't have very good suggestions).

File: falcon/http_error.py [hunk 2]
```diff
@@ -243,6 +238,18 @@ def to_xml(self) -> bytes:
             error_element, encoding='utf-8'
         )
 
+    @deprecated(
+        'The internal serialization to xml is deprecated. A similar output can be '
+        'obtained by serializing to xml the output of ``.to_dict()``'
+    )
+    def to_xml(self) -> bytes:
```
[the remark this answers is not in the document]
Comment (CaselIT) [reply falcon/http_error.py:245]: I don't want the error serializer raising a deprecation warning, to the underscore version is needed

File: docs/_newsfragments/2355.newandimproved.rst [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Added a new flag :attr:`~falcon.ResponseOptions.enable_xml_error_serialization` that
```
[the remark this answers is not in the document]
Comment (CaselIT) [reply docs/_newsfragments/2355.newandimproved.rst:1]: could work

File: falcon/http_error.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vytas7) [reply falcon/http_error.py:245]: Ah, makes sense!

File: falcon/response.py (hunk 1 above)
Comment (vytas7) [reply falcon/response.py:1399]: "but only if the client prefers" sorry

**Review by vytas7 — APPROVED** [review]
LGTM now, thanks for this improvement :+1:
We do want to get rid of any XML support out of the box, although we may want to add XML handlers in the future, either out of the box, or as a recipe, or an add-on library.

I would still like to polish docstrings and the proposed newsfragment, but I can try to address that in a followup PR.

**Edit**: got the ball rolling in https://github.com/falconry/falcon/pull/2358.

**Pull request merged**